arangod/Cluster/ClusterInfo.cpp

```diff
@@ -6215,19 +6215,61 @@ ClusterInfo::getResponsibleServerReplication2(std::string_view shardID) {
     }
   }
 
-  auto logId = LogicalCollection::shardIdToStateId(shardID);
   auto result = std::shared_ptr<std::vector<ServerID>>{nullptr};
 
   while (true) {
     {
+      if (++tries >= 100 || _server.isStopping()) {
+        break;
+      }
+
```

Comment:
Should we introduce some kind of delay / backoff here?
The continue loop here is quite short.
We free the lock,
We add a number,
We ask for the lock.
=> I think there could be some chance that the writeLock never gets in between to update protected data.

Should we add something like (sleep 0.1 * planUpdateInteraval) for every consecutive try to increase the chances for write lock?

Reply:
In General I would prefer if we could get away without busy looping here in the first place.

Reply:
Why is the retry actually required?
Can't we proof that you only get the shardID in your hands, after all _planXXX entries have seen it?

Reply:
There is a sleep at line 6301, for 500ms. Until now, I never experienced problems with this sleep interval. Nevertheless, if you think it's worth it, we could increase it with each iteration.
The retry was there originally and it is required to make sure that leadership has been established, similar to `getResponsibleServerReplication1`, which also has a retry.
Perhaps some of the `continue` statements could be eliminated, but other than that I don't think there's a way to ensure that the log has a leader, apart from waiting.